tests/multicomponent_compressible_projected_density_box.prm

```diff
@@ -0,0 +1,151 @@
+# This is a testcase that demonstrates the projected density approximation within 
+# The multicomponent compressible material model.
+# The setup is a 1x1 m, 2x2 grid where we apply a surface pressure equal 
+# to 50 GPa with gravity disabled. This creates a uniform pressure field that 
+# results in a uniform density distribution. 
+# This allows for no material to flow into/out of the model while allowing 
+# density to accomodate the increased pressure.
+# 
+# The density within multicomponent compressible is determined using the following parameters:
+
+# reference_thermal_expansivities              = 2e-5
+# reference_isothermal_compressibilities       = 4e-12
+# temperature                                  = 1600
+# reference_temperatures                       = 1600
+# isothermal_bulk_modulus_pressure_derivatives = 4
+# reference_densities                          = 3300
+# pressure                                     = 5e10
+# Which are plugged into the following expression for the density:
+
+# ak = reference_thermal_expansivities/reference_isothermal_compressibilities
+# f  = (1. + (pressure - ak*(temperature - reference_temperatures)) *
+#                    isothermal_bulk_modulus_pressure_derivatives   *
+#                          reference_isothermal_compressibilities)
+# densities[c] = reference_densities*std::pow(f, 1./isothermal_bulk_modulus_pressure_derivatives)
+
+# Which should result in a density of 3822.3642114512877, and as we are only
```

Comment:
One suggestion - In other test PRM files I often include complex equations and values that were defined in python. That way, the user can directly copy them into a python interpreter or script and reproduce the values quickly. Maybe consider using that approach above?